src/test/java/com/device/shop/test/controller/ProductControllerTest.java

```diff
@@ -64,6 +79,26 @@ public void testGetAllProducts() throws Exception{
                 .andExpect(jsonPath("$[1].description").value("*"))
                 .andExpect(jsonPath("$[1].sku").value("one"));
     }
+
+    @Test
+    public void testUploadFile() throws Exception {
+        MultipartFile file = Mockito.mock(MultipartFile.class);
+        when(file.getOriginalFilename()).thenReturn("test.csv");
+
+        ResponseEntity<String> response = productController.uploadFile(file);
+
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertEquals("Uploaded the file successfully: test.csv", response.getBody());
+        verify(productService, times(1)).save(file);
     }
+    @Test
+    public void testSave_InvalidCSVFormat() throws Exception{
+        MockMultipartFile file = new MockMultipartFile("file", "test.csv", "text/csv", "invalid csv".getBytes());
 
+        ResultMatcher expectedStatus = status().isOk();
```

Comment:
Invalid test - mock service that throws BadRequestException and expected status should be also bad request, not Ok